### DISCUSSION TITLE
RFE: conversions between old and new format?

### DISCUSSION THREAD

**Discussion by pirat89** [asciinema/discussions#50] [General]
Hi guys,
I realized you are using the new format to store the asciicast. I believe it's welcomed by many people, but the new format is too unfriendly when you want to modify the stored asciicast. Big advantage of the old format was that I have been allowed to control speed of actions that happend during the recording pretty easily (just write command and modify hundreds lines to accelerate too slow parts) or remove everything what I wanted and I had video presentation prepared in short time. However in the new format you are using relative timestamps since the beginning of the recordings, so when I remove something or want to speedup specific part, I have to recalculate all times in the lines below.

Would it be possible to enable recording in the old format? Or better, would it be possible to provide conversions between the old & new format, so I would be able to convert the asciicast to the old format, modify it and then convert it back, so there will not be required modifications in current functionality and just there will be added two more commands in the asciicast? I can create for myself simple script that will do just changes I need (time since the start of recording <-> duration of each item) but I believe functionality like that would be helpful for more people so would be nice to have it integrated.

I will be glad for any solution you possibly integrate that will make modifications of recorded asciicasts simple for people.

**Comment by coryallen83** [comment]
So sorry I will not be requesting your help. How long does it have to stay broken. Am sorry for trying to make to live on my CPU.

**Comment by endreszabo** [comment]
stop whining and do the math.

or use my script to convert timestamps between absolut/relative times: 
https://gist.github.com/endreszabo/3f7e95ee0a1f21fac16b7723e193852b

**Comment by pirat89** [comment]
@endreszabo: That's what I did yesterday.

@coryallen83: Not sure whether I understand you or what to think about your comment. I don't know what will be best solution in the tool from the point of view of developers of the tool or whether they will want to integrate such functionality. That's why I created the rfe with description how I am usually working with the created asciicast and what are current obstacles. Even when I have script for myself that will help me to work with it as before, it doesn't have to be convenient/suitable solution for everyone and someone would be more happy about a different integrated solution.

**Comment by remram44** [comment]
There is also [a script to convert scriptreplay to asciinema v1](https://gist.github.com/izabera/223f58d20be61911e62b), but no way to convert to v2 (or v1 to v2). This is a problem.

**Comment by ku1ik** [comment]
Maybe this will be handy: https://github.com/asciinema/discussions/issues/38

**Comment by ku1ik** [comment]
FYI the new asciinema CLI 3.0, which has been released recently, uses new [asciicast v3](https://docs.asciinema.org/manual/asciicast/v3/) (relative timestamps), and provides `convert` command for file format conversions. No conversion to v1 is supported but v3 is better anyway.
